src/mpid/ch4/netmod/ucx/ucx_win.c

```diff
@@ -68,9 +68,14 @@ static int win_allgather(MPIR_Win * win, size_t length, uint32_t disp_unit, void
         MPIDI_UCX_CHK_STATUS(status);
     }
 
+    MPI_Datatype aint_type;
```

Comment:
I think we can just use `MPI_AINT`.

Reply:
Aside from that, I don't really understand what changed here. `rkey_sizes` is the data. It shouldn't have anything to do with large count interfaces.

Reply:
Oh, I see it now. It is for the followup collective using Allgatherv. This makes sense now, but we should still just use `MPI_AINT` directly.

Reply:
And I suppose all this is tied to the v-collective promotion, since that is the commit that actually promotes the `rkey_sizes` type.

Reply:
> Oh, I see it now. It is for the followup collective using Allgatherv. This makes sense now, but we should still just use `MPI_AINT` directly.

We are. The problem is `MPI_Aint` is not a datatype, so we need create a `MPI_Datatype` for it.

Reply:
`MPI_AINT` is a datatype. https://github.com/pmodels/mpich/blob/e149d18d8cd3047920da4ebab2588c4d1129cf59/src/include/mpi.h.in#L262

Reply:
Great! Yeah, I can use `MPI_AINT` directly. I had a feeling I was missing something.